src/extractor/guidance/turn_handler.cpp

```diff
@@ -90,25 +91,37 @@ Intersection TurnHandler::handleThreeWayTurn(const EdgeID via_edge, Intersection
     const auto isObviousOfTwo = [this](const ConnectedRoad road, const ConnectedRoad other) {
         const auto first_class =
             node_based_graph.GetEdgeData(road.turn.eid).road_classification.road_class;
-        const bool is_ramp = isRampClass(first_class);
+
         const auto second_class =
             node_based_graph.GetEdgeData(other.turn.eid).road_classification.road_class;
-        const bool is_narrow_turn =
-            angularDeviation(road.turn.angle, STRAIGHT_ANGLE) < NARROW_TURN_ANGLE;
-        const bool other_turn_is_at_least_orthogonal =
-            angularDeviation(other.turn.angle, STRAIGHT_ANGLE) > 85;
-        const bool turn_is_perfectly_straight = angularDeviation(road.turn.angle, STRAIGHT_ANGLE) <
-                                                std::numeric_limits<double>::epsilon();
+
+        const bool is_ramp = isRampClass(first_class);
         const bool is_obvious_by_road_class =
             (!is_ramp && (2 * getPriority(first_class) < getPriority(second_class))) ||
             (!isLowPriorityRoadClass(first_class) && isLowPriorityRoadClass(second_class));
+
+        if (is_obvious_by_road_class)
+            return true;
+
+        const bool other_is_obvious_by_road_flass =
+            (!isRampClass(second_class) &&
+             (2 * getPriority(second_class) < getPriority(first_class))) ||
```

Comment:
This is a decision that needs to be moved out into the profile at some point. Goal of further refactoring. The reason behind this is that you don't want to announce a turn if you are on a primary road that turns slightly and a small road exits at that position.